Next new project

benabbon opened the pull request:
Handle the creation of a new dahu project, initializes the models and the screen and writes the new file on the hard drive.

benabbon committed:
introduce a utility function in path :

getDahuFileFromDirectory : Get the name of the presentation file to
create when we create a new project, it is calculated from the name of the Directory.

benabbon committed:
introduce a Directory chooser window in the kernel

benabbon committed:
add the new project button in the menu

benabbon committed:
Handle the creation of a new project

benabbon committed:
Fortify conditions on screen view :

This is important for when we want to create a new project
with no screens on screencastn or when we initialize the workspace with
no screen.

barraq commented:
Good work @benabbon. Please could you take into account my two remarks before I do the merge. I merged this branch with  the new Next (with the Screencast controller) so that you can do the changes. Thanks

barraq committed:
Merge branch 'next' into next-new-project

Conflicts:
	dahu/core/app/scripts/dahuapp.js
	dahu/editor/src/main/java/io/dahuapp/editor/DahuApp.java

benabbon committed:
migrate utility functions to screencast controller

the function moved here is getDahuFileFromDirectory, used
to get the name of the presentation to create relating to
the given directory

benabbon committed:
factorize layout creation code

this code is used when we open an existing project
or when we create a new empty one.

benabbon commented:
They are corrected and pushed in this branch.
